### DISCUSSION TITLE
@babel/parser in browser

### DISCUSSION THREAD

**Discussion by k644606347** [babel/babel#14853] [Q&A] [answered]
I try to run @babel/parser ^ 7.18.9 in the browser to parse strings and display ast. I use webpack to build applications. It works well, but what is the problem with this?

I want the CDN version of @babel/parser, but the official only provides CJS. Maybe the official does not recommend running @babel/parser in the browser?

**Comment by nicolo-ribaudo** [comment, marked as answer]
You can load it from the skypack CDN:
```js
import { parse } from "https://cdn.skypack.dev/@babel/parser"
```
or
```js
const { parse } = await import("https://cdn.skypack.dev/@babel/parser")
```

**Reply by k644606347** [reply]
> You can load it from the skypack CDN:
> 
> ```js
> import { parse } from "https://cdn.skypack.dev/@babel/parser"
> ```
> 
> or
> 
> ```js
> const { parse } = await import("https://cdn.skypack.dev/@babel/parser")
> ```

cool, but I want to download files and upload min.js to my production server.
I plan to customize a global variable such as ```window.__myparser```, and then configure ```externals``` in webpack

**Reply by k644606347** [reply]
> You can load it from the skypack CDN:
> 
> ```js
> import { parse } from "https://cdn.skypack.dev/@babel/parser"
> ```
> 
> or
> 
> ```js
> const { parse } = await import("https://cdn.skypack.dev/@babel/parser")
> ```

I found that Babel provides browser side installation:
```<script src=" https://unpkg.com/ @babel/standalone/babel.min.js"></script>```
can I use ```parse``` through ```window.Babel```?

**Reply by kungfooman** [reply]
@k644606347 No, you can't, it's not exported in `@babel/standalone` (it is *in* the file, but out of reachable scope without a manual rewrite). But what you *can* do is:

```html
<script>
    window.exports = {};
</script>
<script src="./node_modules/@babel/parser/lib/index.js"></script>
```

Don't forget:

```sh
npm install @babel/parser
```

Test it:

![image](https://github.com/babel/babel/assets/5236548/563693b9-e711-4395-aff6-3efa44bc41d6)

**Comment by shijiatongxue** [comment]
same issue

**Comment by shijiatongxue** [comment]
can we use babel parse function by import only babel standalone？

**Reply by kungfooman** [reply]
no, but there is another way to get access to `parse`, see answer above
